docs/reference/indices/flush.asciidoc

```diff
@@ -39,7 +40,152 @@ or even on `_all` the indices.
+recovery without the synced flush marker would take a long time.
+
+To check whether a shard has a marker or not, one can use the `commit` section of shard stats returned by
+the <<indices-stats,indices stats>> API:
+
+[source,bash]
+--------------------------------------------------
+GET /twitter/_stats/commit?level=shards
+--------------------------------------------------
+// AUTOSENSE
+
+[float]
+=== Synced Flush API
+
+The Synced Flush API allows an administrator to initiate a synced flush manually. This can particularly useful for
+a planned (rolling) cluster restart where one can stop indexing and doesn't want to wait for the default 30m to pass
+when the synced flush will be performed automatically.
+
+While handy, there are a couple of caveats for this API:
+
+1. Synced flush is a best effort operation. Any ongoing indexing operations will cause
+the synced flush to fail. This means that some shards may be synced flushed while others aren't. See below for more.
+2. The `sync_id` marker is removed as soon as the shard is flushed again. Uncommitted
+operations in the transaction log do not remove the marker. That is because the marker is store as part
+of a low level lucene commit, representing a point in time snapshot of the segments. In practice, one should consider
+any indexing operation on an index  as removing the marker.
+
+
+[source,bash]
+--------------------------------------------------
+POST /twitter/_flush/synced
+--------------------------------------------------
+// AUTOSENSE
+
+The response contains details about how many shards were successfully synced-flushed and information about any failure.
+
+Here is what it looks like when all shards of a two shards and one replica index successfully
+sync-flushed:
+
+[source,js]
+--------------------------------------------------
+{
+   "_shards": {
+      "total": 4,
+      "successful": 4,
+      "failed": 0
+   },
+   "twitter": {
+      "total": 4,
+      "successful": 4,
+      "failed": 0
+   }
+}
+--------------------------------------------------
+
+
+Here is what it looks like when one shard group failed due to pending operations:
+
+[source,js]
+--------------------------------------------------
+{
+   "_shards": {
+      "total": 4,
+      "successful": 2,
+      "failed": 2
+   },
+   "twitter": {
+      "total": 4,
+      "successful": 2,
+      "failed": 2,
+      "failures": [
+         {
+            "shard": 1,
+            "reason": "[2] ongoing operations on primary"
+         }
+      ]
+   }
+}
+--------------------------------------------------
+
+
+Sometimes the failures are specific to a shard copy, in which case they will be reported as follows:
+
+[source,js]
+--------------------------------------------------
+{
+   "_shards": {
+      "total": 4,
+      "successful": 1,
+      "failed": 1
+   },
+   "twitter": {
+      "total": 4,
+      "successful": 3,
+      "failed": 1,
+      "failures": [
+         {
+            "shard": 1,
```

Comment:
Presumably the other copies succeeded in this case. Is it a good idea to output these in the same total/successful/failed/failures format? Also, it might be useful to mention what this means from a recovery standpoint. Something like "those copies that failed will not be eligible for fast recovery but those that succeeded still will be."